node/router/src/helpers/cache.rs

```diff
@@ -180,39 +177,40 @@ impl<N: Network> Cache<N> {
         key: K,
         interval_in_secs: i64,
     ) -> usize {
+        // Fetch the current timestamp.
+        let now = OffsetDateTime::now_utc();
+
```

Comment:
Couldn't `now` become stale if placed at the top of this method (and before the `.write()` call)?

Reply:
That value is actually closer to the timestamp of the appearance of the new key, making it more accurate; also, the difference would be miniscule, but any locking time - especially writing - should be limited to a minimum.